### ISSUE TITLE
persistentMap.set() not atomic

### ISSUE THREAD

**Issue by kaytwo** [nanostores/persistent#46]
From the overall documentation I would expect that calling `set()` on a persistentMap would be atomic, and none of the listeners would fire until after the entire `set()` call completed. I see that is not the case, and that under the hood there are individual calls to `setKey()`, thus my subscribe logic gets called multiple times as each key gets updated.

Is there some way to "disable interrupts" before running `set()`? Is this a "wontfix" and I should expect to either use the primitive store with json encode/decode as in the documentation so that everything is monolithic? If that is the case, the documentation could use a warning that `persistentMap.set` will cause multiple notify calls to listeners.

**Comment by ai** [comment]
Yes, let’s unify API of `nanostores/map` and `persistentMap` here.

1. Can you double-check, that `map.set()` calls listener only once?
2. Can you send PR and try to not bloar JS bundle size too much?
